src/AppBundle/Tests/Acceptance/ApiContext.php

```diff
@@ -0,0 +1,274 @@
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace AppBundle\Tests\Acceptance;
+
+use AppBundle\DataFixtures\ORM\AdminFixture;
+use AppBundle\DataFixtures\ORM\RoleFixture;
+use AppBundle\DataFixtures\ORM\UserFixture;
+use Assert\Assertion;
+use Behat\Gherkin\Node\PyStringNode;
+use Behat\Gherkin\Node\TableNode;
+use Behat\Symfony2Extension\Context\KernelAwareContext;
+use Behat\Symfony2Extension\Context\KernelDictionary;
+use Symfony\Bundle\FrameworkBundle\Client;
+use Symfony\Component\PropertyAccess\PropertyAccess;
+
+/**
+ * ApiContext.
+ *
+ * Basic behat context which provides main functionality for API testing based on assertions.
+ * This context is capable at configuring a full API request (with authentication if needed) and
+ * to assert against all the required data.
+ *
+ * @author Maximilian Bosch <maximilian.bosch.27@gmail.com>
+ */
+class ApiContext implements KernelAwareContext
+{
+    use KernelDictionary;
+
+    /**
+     * @var Client
+     */
+    private $client;
+
+    /**
+     * @var string
+     */
+    private $apiKey;
+
+    /**
+     * @var mixed[]
+     */
+    private $requestPayload = [];
+
+    /**
+     * @var array
+     */
+    private $response;
+
+    /**
+     * @var int
+     */
+    private $responseCode;
+
+    /**
+     * @var bool
+     */
+    private $profilerEnabled = false;
+
+    /**
+     * @var \Symfony\Component\HttpKernel\Profiler\Profile
+     */
+    private $profiler;
+
+    /** @BeforeScenario */
+    public function prepare()
+    {
+        $container = $this->getContainer();
+
+        // loads the client into the context
+        $this->client = $container->get('test.client');
+
+        // re-applies the fixtures
+        $container->get('app.doctrine.fixtures_loader')->applyFixtures([
+            UserFixture::class,
+            RoleFixture::class,
+            AdminFixture::class,
+        ]);
+    }
+
+    /** @AfterScenario */
+    public function cleanUp()
+    {
+        $this->client          = null;
+        $this->apiKey          = null;
+        $this->response        = null;
+        $this->requestPayload  = [];
+        $this->responseCode    = null;
+        $this->profiler        = null;
+        $this->profilerEnabled = false;
+    }
+
+    /**
+     * @Given /^I'm authenticated as "(.*)" with password "(.*)"$/
+     *
+     * @param string $username
+     * @param string $password
+     */
+    public function authenticate(string $username, string $password)
+    {
+        $this->client->request('POST', '/api/api-key.json', ['login' => $username, 'password' => $password]);
+        $response = $this->client->getResponse();
+
+        Assertion::eq(200, $response->getStatusCode(), sprintf(
+            'Authentication with username "%s" and password "%s" failed!',
+            $username,
+            $password
+        ));
+
+        $this->apiKey = json_decode($response->getContent(), true)['apiKey'];
```

Comment:
use `$this->decode($json)` here (might be more readable than `json_decode(..., true)` and covers edge cases with a broken response